azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerPipelineTest.java

```diff
@@ -499,164 +383,73 @@ public void testBasicPipelineLevel2Run2() throws Exception {
-    compareStates(previousExpectedStateMap, previousNodeMap);
-    compareStates(pipelineExpectedStateMap, pipelineNodeMap);
+    assertStatus(pipelineFlow, "pipeline1_1", Status.SUCCEEDED);
+    assertStatus(pipelineFlow, "pipeline1_1:innerFlow2", Status.SUCCEEDED);
+    assertStatus(pipelineFlow, "pipeline1_2", Status.RUNNING);
+    assertStatus(pipelineFlow, "pipeline1_2:innerJobA", Status.QUEUED);
 
     InteractiveTestJob.getTestJob("prev:pipeline1_2:innerFlow2").succeedJob();
-    pause(250);
-    previousExpectedStateMap.put("pipeline1_2:innerFlow2", Status.SUCCEEDED);
-    previousExpectedStateMap.put("pipeline1_2", Status.SUCCEEDED);
-    Assert.assertEquals(Status.SUCCEEDED, previousFlow.getStatus());
-    Assert.assertFalse(thread1.isAlive());
-    pipelineExpectedStateMap.put("pipeline1_2:innerJobA", Status.RUNNING);
-    compareStates(previousExpectedStateMap, previousNodeMap);
-    compareStates(pipelineExpectedStateMap, pipelineNodeMap);
+    assertStatus(previousFlow, "pipeline1_2:innerFlow2", Status.SUCCEEDED);
+    assertStatus(previousFlow, "pipeline1_2", Status.SUCCEEDED);
+    assertFlowStatus(previousFlow, Status.SUCCEEDED);
+    assertThreadShutDown(previousRunner);
+    assertStatus(pipelineFlow, "pipeline1_2:innerJobA", Status.RUNNING);
 
     InteractiveTestJob.getTestJob("pipe:pipeline1_2:innerJobA").succeedJob();
-    pause(250);
-    pipelineExpectedStateMap.put("pipeline1_2:innerJobA", Status.SUCCEEDED);
-    pipelineExpectedStateMap.put("pipeline1_2:innerFlow2", Status.RUNNING);
-    compareStates(previousExpectedStateMap, previousNodeMap);
-    compareStates(pipelineExpectedStateMap, pipelineNodeMap);
+    assertStatus(pipelineFlow, "pipeline1_2:innerJobA", Status.SUCCEEDED);
+    assertStatus(pipelineFlow, "pipeline1_2:innerFlow2", Status.RUNNING);
 
     InteractiveTestJob.getTestJob("pipe:pipeline1_2:innerFlow2").succeedJob();
-    pause(250);
-    pipelineExpectedStateMap.put("pipeline1_2", Status.SUCCEEDED);
-    pipelineExpectedStateMap.put("pipeline1_2:innerFlow2", Status.SUCCEEDED);
-    compareStates(previousExpectedStateMap, previousNodeMap);
-    compareStates(pipelineExpectedStateMap, pipelineNodeMap);
-    Assert.assertEquals(Status.SUCCEEDED, pipelineFlow.getStatus());
-    Assert.assertFalse(thread2.isAlive());
+    assertStatus(pipelineFlow, "pipeline1_2", Status.SUCCEEDED);
+    assertStatus(pipelineFlow, "pipeline1_2:innerFlow2", Status.SUCCEEDED);
+    assertFlowStatus(pipelineFlow, Status.SUCCEEDED);
+    assertThreadShutDown(pipelineRunner);
   }
 
-  private Thread runFlowRunnerInThread(final FlowRunner runner) {
+  private void runFlowRunnerInThread(final FlowRunner runner) {
     final Thread thread = new Thread(runner);
     thread.start();
-    return thread;
-  }
-
-  private void pause(final long millisec) {
-    try {
-      Thread.sleep(millisec);
-    } catch (final InterruptedException e) {
-    }
-  }
-
-  private void createExpectedStateMap(final ExecutableFlowBase flow,
```

Comment:
I really don't see why. After all, `compareStates` only checked for the "expected states" only. Maybe there was a bit more control on something like "checking again that statuses haven't changed on execution 1 although expected them to change on execution 2". But when I was converting these to single line asserts, I didn't encounter any case where it would've seemed important to "check again that statuses haven't changed". So I think it was originally just done in an over-complicated way "by accident" :)